### PR TITLE
Fixed issue #59.  Changed the create team page into a dialog window.

### PR DESCRIPTION
**Pull request by cdterry87**
Also removed the old way of creating teams.

### CODE REVIEW

**Comment by chamerling** [comment]
@cdterry87 dialog remains displayed after creation on all pages. Please clean the state

Capture d’écran 2019-10-06 à 11 42 47

**Review by chamerling — COMMENTED** [review]
Looks good, just fix the dialog close bug and we are good

**Comment by cdterry87** [comment]
@chamerling I am now handling the state of the Create Team dialog in state and clearing it out after a team is created.  Check it out now and let me know if there are any issues.

File: src/App.vue [hunk 1]
```diff
@@ -148,6 +148,19 @@
...
+      <v-dialog v-model="createTeamDialog" width="500">
+          <v-card>
+            <v-toolbar dark>
+                <v-btn icon dark @click="createTeam">
+                <v-icon>close</v-icon>
+                </v-btn>
+                <v-toolbar-title class="white--text">Create a new team</v-toolbar-title>
+            </v-toolbar>
+            <v-card-text>
+              <CreateTeam />
```
Comment (chamerling) [inline src/App.vue]: All the things with vuex state to manage `createTeamDialog` is too complex. You should just be able to manage local state in this App component. Add an event listener to `CreateTeam` which emits event when form is submitted and then hide the dialog from current component

```html
<CreateTeam @close="dialog = false"/>
```

File: src/components/CreateTeam.vue [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          name: this.name,
+          icon: this.icon,
+          usernames: this.members.map(member => member.username)
+        };
+
+        this.$store.dispatch("createTeam", team);
+        this.$store.dispatch("setCreateTeamDialog", false);
+        this.$router.push({ name: "team", params: { name: team.name } });
+
+        this.clear();
```
Comment (chamerling) [inline src/components/CreateTeam.vue:96]: Then here 

```js
this.$emit("close")
```

**Comment by cdterry87** [comment]
@chamerling I was definitely overthinking this, lol.  I implemented your solution, and it is working great now.  I removed everything related to the dialog from the vuex store.  Thank you for the feedback and the help getting this resolved!  Let me know if there is anything else I can do to help.

**Comment by chamerling** [comment]
@cdterry87 great!

**Pull request merged**